Memory leak in TCP binding

Hey there. I have the classy problem of openhab dying after some hours and could break it down to one addon:

Here, have a look at my config:

``` ini
# openhab.cfg
################################# TCP - UDP Binding ###################################
tcp:retryinterval=60
tcp:queue=false
tcp:refreshinterval=2000
```

``` java
// items
String AACC              "LEDs"  <colorwheel> (Lichter)    { tcp=">[192.168.1.100:2323:'REGEX((.*))']" }
String Leroy_X10receiver "X10Receiver.net [%s]"  /**/ /**/ { tcp=">[192.168.1.100:9000:'REGEX((.*))']" }
```
## Reproduction
- Listen to TCP 9000 and 2323 on 192.168.1.100
- Openhab ESTABLISHES TCP connection

_(Everything works fine and WILL stay fine here)_
- Break connection by putting 192.168.1.100 to STANDBY (Is Windows)
- Connections time out with CLOSE_WAIT
- 192.168.1.100 will never answer with a CLOSING or CLOSED
- TCP Binding stays stuck in a loop and is spamming:

```
[...]
The channel for /192.168.1.100:2323 is now connected
The channel for /192.168.1.100:9000 is now connected
The channel for /192.168.1.100:2323 is now connected
The channel for /192.168.1.100:9000 is now connected
The channel for /192.168.1.100:2323 is now connected
The channel for /192.168.1.100:9000 is now connected
[...]
```

_(repeats with every worker (refresh) cycle)_

Open file descriptors rise to a non-critical value. Memory rises to the limits of Xmx Java Heap Space (but not to the system's limits) until it finally breaks:

```
2015-06-06 05:19:36.108 [INFO ] [t.AbstractSocketChannelBinding] - The channel for /192.168.1.100:2323 is now connected
2015-06-06 05:19:36.556 [INFO ] [t.AbstractSocketChannelBinding] - The channel for /192.168.1.100:2323 is now connected
2015-06-06 05:19:37.010 [INFO ] [t.AbstractSocketChannelBinding] - The channel for /192.168.1.100:2323 is now connected
Exception in thread "fileinstall-addons" Exception in thread "qtp14920870-44" java.lang.OutOfMemoryError: Java heap space
        at org.eclipse.jetty.util.thread.QueuedThreadPool.newThread(QueuedThreadPool.java:450)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.startThread(QueuedThreadPool.java:430)
        at org.eclipse.jetty.util.thread.QueuedThreadPool.dispatch(QueuedThreadPool.java:361)
        at org.eclipse.jetty.server.nio.SelectChannelConnector$ConnectorSelectorManager.dispatch(SelectChannelConnector.java:295)
        at org.eclipse.jetty.io.nio.SelectorManager$SelectSet.doSelect(SelectorManager.java:703)

```

By the way - the binding won't re-connect the tcp socket, even if I start 192.168.100 again. It just keeps spamming.
Without the TCP Binding the system ran fine multiple days - although being some kind of useless, of course ;)

If I can provide you anything more, please tell me! Here are some graphs. From the logs you can see it broke at about 05:19 AM.

(timestamps from log and graphs may differ in timezone (was UTC instead of CEST)

![cpul](https://cloud.githubusercontent.com/assets/2018039/8041451/8e70da14-0e18-11e5-85e5-5c6c8351e34f.png)
![cpuu](https://cloud.githubusercontent.com/assets/2018039/8041452/8e756cbe-0e18-11e5-8303-8a2e3ccc1e2f.png)
![mem](https://cloud.githubusercontent.com/assets/2018039/8041453/8e75f9b8-0e18-11e5-93a9-f8a7cc3ddfdc.png)
![openfd](https://cloud.githubusercontent.com/assets/2018039/8041454/8e770cf4-0e18-11e5-8ddf-0663b1b1b697.png)
![tcp](https://cloud.githubusercontent.com/assets/2018039/8041455/8e811eec-0e18-11e5-88c5-08f095a651b8.png)
##

I'm having the same issue with my Marantz amplifier TCP binding. Just got the amplifier to power off by TCP packet. TCP binding seems to uninstall itself after daemon restart.

openhab.log
`2017-04-21 18:05:51.061 [INFO ] [b.core.service.AbstractActiveService] - TCP Refresh Service has been shut down
2017-04-21 18:05:51.659 [INFO ] [ui.habmin.internal.servlet.HABminApp] - Stopped HABmin servlet
2017-04-21 18:05:51.793 [ERROR] [org.quartz.core.JobRunShell         ] - Job org.openhab.binding.tcp.protocol.internal.TCPBinding@42406af6.42406af6-Reconnect-1492797946786 threw an unhandled Exception: 
java.nio.channels.ClosedSelectorException
	at sun.nio.ch.EPollSelectorImpl.implRegister(EPollSelectorImpl.java:174)[:1.8.0_121]
	at sun.nio.ch.SelectorImpl.register(SelectorImpl.java:132)[:1.8.0_121]
	at java.nio.channels.spi.AbstractSelectableChannel.register(AbstractSelectableChannel.java:212)[:1.8.0_121]
	at java.nio.channels.SelectableChannel.register(SelectableChannel.java:280)[:1.8.0_121]
	at org.openhab.binding.tcp.AbstractSocketChannelBinding$ReconnectJob.execute(AbstractSocketChannelBinding.java:1150)[229:org.openhab.binding.tcp:1.10.0.201704100111]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)[105:org.eclipse.smarthome.core.scheduler:0.9.0.201704011656]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)[105:org.eclipse.smarthome.core.scheduler:0.9.0.201704011656]
2017-04-21 18:05:51.802 [ERROR] [org.quartz.core.ErrorLogger         ] - Job (org.openhab.binding.tcp.protocol.internal.TCPBinding@42406af6.42406af6-Reconnect-1492797946786 threw an exception.
org.quartz.SchedulerException: Job threw an unhandled exception. [See nested exception: java.nio.channels.ClosedSelectorException]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:213)[105:org.eclipse.smarthome.core.scheduler:0.9.0.201704011656]
	at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)[105:org.eclipse.smarthome.core.scheduler:0.9.0.201704011656]
Caused by: java.nio.channels.ClosedSelectorException
	at sun.nio.ch.EPollSelectorImpl.implRegister(EPollSelectorImpl.java:174)[:1.8.0_121]
	at sun.nio.ch.SelectorImpl.register(SelectorImpl.java:132)[:1.8.0_121]
	at java.nio.channels.spi.AbstractSelectableChannel.register(AbstractSelectableChannel.java:212)[:1.8.0_121]
	at java.nio.channels.SelectableChannel.register(SelectableChannel.java:280)[:1.8.0_121]
	at org.openhab.binding.tcp.AbstractSocketChannelBinding$ReconnectJob.execute(AbstractSocketChannelBinding.java:1150)[229:org.openhab.binding.tcp:1.10.0.201704100111]
	at org.quartz.core.JobRunShell.run(JobRunShell.java:202)[105:org.eclipse.smarthome.core.scheduler:0.9.0.201704011656]
	... 1 more
2017-04-21 18:05:52.916 [INFO ] [arthome.ui.paper.internal.PaperUIApp] - Stopped Paper UI
2017-04-21 18:05:52.944 [INFO ] [panel.internal.HABPanelDashboardTile] - Stopped HABPanel
2017-04-21 18:05:52.963 [INFO ] [.dashboard.internal.DashboardService] - Stopped dashboard
2017-04-21 18:06:27.361 [INFO ] [ore.internal.i18n.LocaleProviderImpl] - Locale set to fi_FI
2017-04-21 18:06:40.041 [INFO ] [el.core.internal.ModelRepositoryImpl] - Loading model 'marantz.items'
2017-04-21 18:06:54.210 [INFO ] [.dashboard.internal.DashboardService] - Started dashboard at /start
2017-04-21 18:06:56.365 [INFO ] [arthome.ui.paper.internal.PaperUIApp] - Started Paper UI at /paperui
2017-04-21 18:06:57.685 [INFO ] [ui.habmin.internal.servlet.HABminApp] - Started HABmin servlet at /habmin
2017-04-21 18:06:58.100 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2017-04-21 18:06:58.414 [WARN ] [g.eclipse.smarthome.core.net.NetUtil] - Found multiple local interfaces - ignoring 192.168.1.214
2017-04-21 18:06:59.202 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Listening for incoming connections on /0:0:0:0:0:0:0:0:24001
2017-04-21 18:06:59.210 [INFO ] [ing.tcp.protocol.internal.TCPBinding] - The preamble for all write operations will be set to the default value of ""
2017-04-21 18:06:59.224 [INFO ] [ing.tcp.protocol.internal.TCPBinding] - The postamble for all write operations will be set to the default value of ""
2017-04-21 18:06:59.225 [INFO ] [b.core.service.AbstractActiveService] - TCP Refresh Service has been started
2017-04-21 18:06:59.244 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=kitchen_speakers_device, command=0, direction=OUT, remote=/192.168.1.151:8080, buffer=, isBlocking=false, isReconnecting=false, channel=, host=192.168.1.151, port=8080] 
2017-04-21 18:06:59.246 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=Marantz_ONOFF, command=ON, direction=OUT, remote=/192.168.1.151:8080, buffer=, isBlocking=false, isReconnecting=false, channel=, host=192.168.1.151, port=8080] 
2017-04-21 18:06:59.250 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=MarantzON, command=ON, direction=OUT, remote=/192.168.1.151:8080, buffer=, isBlocking=false, isReconnecting=false, channel=, host=192.168.1.151, port=8080] 
2017-04-21 18:06:59.253 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=speaker_input_bedroom, command=ON, direction=OUT, remote=/10.42.0.168:10006, buffer=, isBlocking=false, isReconnecting=false, channel=, host=10.42.0.168, port=10006] 
2017-04-21 18:06:59.269 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - The channel for /192.168.1.151:8080 is now connected
2017-04-21 18:06:59.558 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - The channel for /192.168.1.151:8080 is now connected
2017-04-21 18:06:59.876 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - The channel for /192.168.1.151:8080 is now connected
2017-04-21 18:07:02.420 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:07:07.439 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:07:14.487 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:07:19.494 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:07:22.652 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:07:27.658 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:07:34.713 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:07:39.720 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:07:43.724 [ERROR] [org.openhab.binding.tcp             ] - [org.openhab.binding.tcp.protocol.udp(156)] The deactivate method has thrown an exception
java.lang.NullPointerException
	at org.openhab.binding.tcp.AbstractDatagramChannelBinding.deactivate(AbstractDatagramChannelBinding.java:731)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)[:1.8.0_121]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)[:1.8.0_121]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[:1.8.0_121]
	at java.lang.reflect.Method.invoke(Method.java:498)[:1.8.0_121]
	at org.apache.felix.scr.impl.inject.BaseMethod.invokeMethod(BaseMethod.java:224)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.inject.BaseMethod.access$500(BaseMethod.java:39)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.inject.BaseMethod$Resolved.invoke(BaseMethod.java:617)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.inject.BaseMethod.invoke(BaseMethod.java:501)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:302)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.inject.ActivateMethod.invoke(ActivateMethod.java:294)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.disposeImplementationObject(SingleComponentManager.java:343)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.SingleComponentManager.deleteComponent(SingleComponentManager.java:152)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.doDeactivate(AbstractComponentManager.java:813)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.deactivateInternal(AbstractComponentManager.java:787)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.DependencyManager.deactivateComponentManager(DependencyManager.java:2225)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.DependencyManager.access$500(DependencyManager.java:55)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.removedService(DependencyManager.java:386)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.DependencyManager$MultipleDynamicCustomizer.removedService(DependencyManager.java:291)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerRemoved(ServiceTracker.java:1241)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.customizerRemoved(ServiceTracker.java:1136)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.ServiceTracker$AbstractTracked.untrack(ServiceTracker.java:996)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:1175)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.BundleComponentActivator$ListenerInfo.serviceChanged(BundleComponentActivator.java:127)[32:org.apache.felix.scr:2.0.6]
	at org.eclipse.osgi.internal.serviceregistry.FilteredServiceListener.serviceChanged(FilteredServiceListener.java:109)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:914)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEventPrivileged(ServiceRegistry.java:862)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistry.publishServiceEvent(ServiceRegistry.java:801)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.serviceregistry.ServiceRegistrationImpl.unregister(ServiceRegistrationImpl.java:222)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.unregister(AbstractComponentManager.java:908)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager$3.unregister(AbstractComponentManager.java:873)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.RegistrationManager.changeRegistration(RegistrationManager.java:139)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.unregisterService(AbstractComponentManager.java:950)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.doDeactivate(AbstractComponentManager.java:805)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.deactivateInternal(AbstractComponentManager.java:787)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.AbstractComponentManager.dispose(AbstractComponentManager.java:579)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.manager.ConfigurableComponentHolder.disposeComponents(ConfigurableComponentHolder.java:706)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.BundleComponentActivator.dispose(BundleComponentActivator.java:523)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.Activator.disposeComponents(Activator.java:452)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.Activator.access$300(Activator.java:54)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.scr.impl.Activator$ScrExtension.destroy(Activator.java:306)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.utils.extender.AbstractExtender$2.run(AbstractExtender.java:290)[32:org.apache.felix.scr:2.0.6]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)[:1.8.0_121]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_121]
	at org.apache.felix.utils.extender.AbstractExtender.destroyExtension(AbstractExtender.java:312)[32:org.apache.felix.scr:2.0.6]
	at org.apache.felix.utils.extender.AbstractExtender.bundleChanged(AbstractExtender.java:186)[32:org.apache.felix.scr:2.0.6]
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:902)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:165)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:75)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:67)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:102)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.Module.publishEvent(Module.java:466)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.Module.doStop(Module.java:624)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.container.Module.stop(Module.java:488)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.eclipse.osgi.internal.framework.EquinoxBundle.stop(EquinoxBundle.java:419)[org.eclipse.osgi-3.10.101.v20150820-1432.jar:]
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.stopBundle(FeaturesServiceImpl.java:1293)[8:org.apache.karaf.features.core:4.0.8]
	at org.apache.karaf.features.internal.service.Deployer.deploy(Deployer.java:612)[8:org.apache.karaf.features.core:4.0.8]
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl.doProvision(FeaturesServiceImpl.java:1176)[8:org.apache.karaf.features.core:4.0.8]
	at org.apache.karaf.features.internal.service.FeaturesServiceImpl$1.call(FeaturesServiceImpl.java:1074)[8:org.apache.karaf.features.core:4.0.8]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)[:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)[:1.8.0_121]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)[:1.8.0_121]
	at java.lang.Thread.run(Thread.java:745)[:1.8.0_121]
2017-04-21 18:07:43.763 [INFO ] [b.core.service.AbstractActiveService] - TCP Refresh Service has been shut down
2017-04-21 18:07:43.962 [INFO ] [core.karaf.internal.FeatureInstaller] - Uninstalled 'openhab-binding-tcp1'
2017-04-21 18:11:00.578 [INFO ] [thome.io.rest.core.item.ItemResource] - Received HTTP POST request at 'items/null' for the unknown item 'null'.
2017-04-21 18:11:40.947 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Listening for incoming connections on /0:0:0:0:0:0:0:0:24001
2017-04-21 18:11:40.949 [INFO ] [b.core.service.AbstractActiveService] - TCP Refresh Service has been started
2017-04-21 18:11:40.949 [INFO ] [ing.tcp.protocol.internal.TCPBinding] - The preamble for all write operations will be set to the default value of ""
2017-04-21 18:11:40.951 [INFO ] [ing.tcp.protocol.internal.TCPBinding] - The postamble for all write operations will be set to the default value of ""
2017-04-21 18:11:40.968 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=kitchen_speakers_device, command=0, direction=OUT, remote=/192.168.1.151:8080, buffer=, isBlocking=false, isReconnecting=false, channel=, host=192.168.1.151, port=8080] 
2017-04-21 18:11:40.971 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=Marantz_ONOFF, command=ON, direction=OUT, remote=/192.168.1.151:8080, buffer=, isBlocking=false, isReconnecting=false, channel=, host=192.168.1.151, port=8080] 
2017-04-21 18:11:40.975 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=MarantzON, command=ON, direction=OUT, remote=/192.168.1.151:8080, buffer=, isBlocking=false, isReconnecting=false, channel=, host=192.168.1.151, port=8080] 
2017-04-21 18:11:40.988 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Connecting the channel Channel [item=speaker_input_bedroom, command=ON, direction=OUT, remote=/10.42.0.168:10006, buffer=, isBlocking=false, isReconnecting=false, channel=, host=10.42.0.168, port=10006] 
2017-04-21 18:11:40.993 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - The channel for /192.168.1.151:8080 is now connected
2017-04-21 18:11:41.032 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - The channel for /192.168.1.151:8080 is now connected
2017-04-21 18:11:41.109 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - The channel for /192.168.1.151:8080 is now connected
2017-04-21 18:11:41.382 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:11:46.392 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:12:17.545 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:12:22.557 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:12:23.593 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:12:28.603 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:12:28.632 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:12:33.643 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:12:33.671 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:12:38.717 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:12:55.365 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:12:55.375 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command 0 on item kitchen_speakers_device  
2017-04-21 18:12:55.381 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:12:55.386 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command ON on item MarantzON  
2017-04-21 18:12:55.392 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:12:55.398 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command ON on item Marantz_ONOFF  
2017-04-21 18:13:09.987 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:14.997 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:13:15.025 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:20.036 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:13:23.081 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:28.091 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:13:28.122 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:13:28.126 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command 0 on item kitchen_speakers_device  
2017-04-21 18:13:28.129 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:28.140 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:13:28.144 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command ON on item MarantzON  
2017-04-21 18:13:28.150 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:13:28.157 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command ON on item Marantz_ONOFF  
2017-04-21 18:13:33.138 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:13:33.191 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:35.468 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:1
' with the file 'marantz.map' : Target value not found in map for '@PWR:1
'
2017-04-21 18:13:35.475 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:1
 to match command 0 on item kitchen_speakers_device  
2017-04-21 18:13:35.480 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:1
' with the file 'marantz.map' : Target value not found in map for '@PWR:1
'
2017-04-21 18:13:35.484 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:1
 to match command ON on item MarantzON  
2017-04-21 18:13:35.490 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:1
' with the file 'marantz.map' : Target value not found in map for '@PWR:1
'
2017-04-21 18:13:35.495 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:1
 to match command ON on item Marantz_ONOFF  
2017-04-21 18:13:38.202 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:13:41.036 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:13:41.040 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command 0 on item kitchen_speakers_device  
2017-04-21 18:13:41.046 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:13:41.051 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command ON on item MarantzON  
2017-04-21 18:13:41.055 [WARN ] [rm.AbstractFileTransformationService] - Could not transform '@PWR:2
' with the file 'marantz.map' : Target value not found in map for '@PWR:2
'
2017-04-21 18:13:41.059 [WARN ] [ing.tcp.protocol.internal.TCPBinding] - Can not parse input @PWR:2
 to match command ON on item Marantz_ONOFF  
2017-04-21 18:13:45.340 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:50.350 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:13:53.396 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:13:58.412 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:05.482 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:10.493 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:13.544 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:18.553 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:18.580 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:23.590 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:23.621 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:28.634 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:28.912 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:33.922 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:33.953 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:38.964 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:38.991 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:44.000 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:44.030 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:49.041 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:14:49.069 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:14:54.078 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:15:01.149 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:15:06.156 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:15:21.262 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:15:26.277 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:15:26.306 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:15:31.314 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:15:31.345 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:15:36.354 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:15:43.428 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:15:48.438 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:16:16.822 [INFO ] [panel.internal.HABPanelDashboardTile] - Stopped HABPanel
2017-04-21 18:16:16.982 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2017-04-21 18:16:19.662 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:16:19.899 [INFO ] [panel.internal.HABPanelDashboardTile] - Stopped HABPanel
2017-04-21 18:16:20.029 [INFO ] [panel.internal.HABPanelDashboardTile] - Started HABPanel at /habpanel
2017-04-21 18:16:24.669 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:16:25.703 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:16:30.711 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:16:33.758 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:16:38.769 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:16:38.799 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:16:43.807 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:16:46.851 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:16:51.861 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
2017-04-21 18:16:54.908 [WARN ] [ing.tcp.AbstractSocketChannelBinding] - The channel java.nio.channels.SocketChannel[closed] has encountered an unknown IO Exception: No route to host
2017-04-21 18:16:59.916 [INFO ] [ing.tcp.AbstractSocketChannelBinding] - Attempting to reconnect the channel for /10.42.0.168:10006
`

--- a/bundles/binding/org.openhab.binding.tcp/src/main/java/org/openhab/binding/tcp/AbstractDatagramChannelBinding.java
+++ b/bundles/binding/org.openhab.binding.tcp/src/main/java/org/openhab/binding/tcp/AbstractDatagramChannelBinding.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2010-2016 by the respective copyright holders.
+ * Copyright (c) 2010-2017 by the respective copyright holders.
  *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -191,7 +191,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                 response = response + "]";
                 return response;
             } catch (Exception e) {
-                logger.error("An exception occurred while converting Channel to String {}", e.getMessage());
+                logger.warn("An exception occurred while converting Channel to String {}", e.getMessage());
             }
             return "";
         }
@@ -692,11 +692,11 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                 try {
                     listenerKey = listenerChannel.register(selector, listenerChannel.validOps());
                 } catch (ClosedChannelException e1) {
-                    logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+                    logger.warn("An exception occurred while registering a selector: {}", e1.getMessage());
                 }
             }
         } catch (Exception e3) {
-            logger.error("An exception occurred while creating the Listener Channel on port number {} ({})",
+            logger.warn("An exception occurred while creating the Listener Channel on port number {} ({})",
                     listenerPort, e3.getMessage());
         }
     }
@@ -711,7 +711,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
         try {
             selector = Selector.open();
         } catch (IOException e) {
-            logger.error("An exception occurred while registering the selector: {}", e.getMessage());
+            logger.warn("An exception occurred while registering the selector: {}", e.getMessage());
         }
     }
 
@@ -724,13 +724,15 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
         try {
             selector.close();
         } catch (IOException e) {
-            logger.error("An exception occurred while closing the selector: {}", e.getMessage());
+            logger.warn("An exception occurred while closing the selector: {}", e.getMessage());
         }
 
         try {
-            listenerChannel.close();
+            if (listenerChannel != null) {
+                listenerChannel.close();
+            }
         } catch (IOException e) {
-            logger.error("An exception occurred while closing the Listener Channel on port number {} ({})",
+            logger.warn("An exception occurred while closing the Listener Channel on port number {} ({})",
                     listenerPort, e.getMessage());
         }
     }
@@ -917,7 +919,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                         try {
                             scheduler = StdSchedulerFactory.getDefaultScheduler();
                         } catch (SchedulerException e1) {
-                            logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                            logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                     e1.getMessage());
                         }
 
@@ -943,7 +945,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                 }
                             }
                         } catch (SchedulerException e) {
-                            logger.error("An exception occurred while scheduling a job with the Quartz Scheduler {}",
+                            logger.warn("An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                     e.getMessage());
                         }
                     }
@@ -958,7 +960,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                         }
                     }
                 } else {
-                    logger.error("there is no channel that services [itemName={}, command={}]", itemName, command);
+                    logger.warn("There is no channel that services [itemName={}, command={}]", itemName, command);
                 }
             }
         }
@@ -1102,13 +1104,13 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                         try {
                             theChannel.channel.close();
                         } catch (IOException e) {
-                            logger.error("An exception occurred while closing a channel: {}", e.getMessage());
+                            logger.warn("An exception occurred while closing a channel: {}", e.getMessage());
                         }
 
                         try {
                             theChannel.channel = DatagramChannel.open();
                         } catch (IOException e) {
-                            logger.error("An exception occurred while opening a channel: {}", e.getMessage());
+                            logger.warn("An exception occurred while opening a channel: {}", e.getMessage());
                         }
 
                         theChannel.isBlocking = false;
@@ -1118,7 +1120,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                             theChannel.channel.configureBlocking(false);
                             // setKeepAlive(true);
                         } catch (Exception e) {
-                            logger.error("An exception occurred while configuring a channel: {}", e.getMessage());
+                            logger.warn("An exception occurred while configuring a channel: {}", e.getMessage());
                         }
 
                         synchronized (theBinding.selector) {
@@ -1128,7 +1130,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                     theChannel.channel.register(theBinding.selector, theChannel.channel.validOps());
                                 }
                             } catch (ClosedChannelException e1) {
-                                logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+                                logger.warn("An exception occurred while registering a selector: {}", e1.getMessage());
                             }
                         }
 
@@ -1156,7 +1158,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                 logger.info("Attempting to reconnect the channel for {}", theChannel.remote);
                             }
                         } catch (Exception e) {
-                            logger.error("An exception occurred while connecting a channel: {}", e.getMessage());
+                            logger.warn("An exception occurred while connecting a channel: {}", e.getMessage());
                         }
                     } else {
                         logger.warn("The listener channel can not be closed!");
@@ -1287,7 +1289,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                             boolean assigned = false;
 
                             if (useAddressMask && (remoteHost.equals("*") || remotePort.equals("*"))) {
-                                logger.error(
+                                logger.warn(
                                         "We do not accept outgoing connections for Items that do use address masks");
                             } else {
 
@@ -1337,7 +1339,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                         try {
                                             newDatagramChannel = DatagramChannel.open();
                                         } catch (IOException e2) {
-                                            logger.error("An exception occurred while opening a channel: {}",
+                                            logger.warn("An exception occurred while opening a channel: {}",
                                                     e2.getMessage());
                                         }
 
@@ -1345,7 +1347,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                             newDatagramChannel.configureBlocking(false);
                                             // setKeepAlive(true);
                                         } catch (IOException e) {
-                                            logger.error("An exception occurred while configuring a channel: {}",
+                                            logger.warn("An exception occurred while configuring a channel: {}",
                                                     e.getMessage());
                                         }
 
@@ -1354,7 +1356,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                             try {
                                                 newDatagramChannel.register(selector, newDatagramChannel.validOps());
                                             } catch (ClosedChannelException e1) {
-                                                logger.error("An exception occurred while registering a selector: {}",
+                                                logger.warn("An exception occurred while registering a selector: {}",
                                                         e1.getMessage());
                                             }
                                         }
@@ -1366,7 +1368,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                             logger.info("'Connecting' the channel {} ", newChannel);
                                             newDatagramChannel.connect(remoteAddress);
                                         } catch (IOException e) {
-                                            logger.error("An exception occurred while connecting a channel: {}",
+                                            logger.warn("An exception occurred while connecting a channel: {}",
                                                     e.getMessage());
                                         }
                                     } else {
@@ -1387,7 +1389,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                 // Wait for an event
                 selector.selectNow();
             } catch (IOException e) {
-                logger.error("An exception occurred while Selecting ({})", e.getMessage());
+                logger.warn("An exception occurred while Selecting ({})", e.getMessage());
             }
         }
 
@@ -1436,7 +1438,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                 logger.warn("The channel for {} has no connection pending ({})",
                                         theDatagramChannel.getRemoteAddress(), e.getMessage());
                             } catch (IOException e1) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while getting the remote address of channel {} ({})",
                                         theDatagramChannel, e1.getMessage());
                             }
@@ -1447,7 +1449,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                 logger.warn("The channel for {} has encountered an unknown IO Exception: {}",
                                         theDatagramChannel.getRemoteAddress(), e.getMessage());
                             } catch (IOException e1) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while getting the remote address of channel {} ({})",
                                         theDatagramChannel, e1.getMessage());
                             }
@@ -1465,7 +1467,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                 logger.warn("The channel for {} is closed ({})", theDatagramChannel.getRemoteAddress(),
                                         e.getMessage());
                             } catch (IOException e1) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while getting the remote address of channel {} ({})",
                                         theDatagramChannel, e1.getMessage());
                             }
@@ -1480,7 +1482,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                             try {
                                 scheduler = StdSchedulerFactory.getDefaultScheduler();
                             } catch (SchedulerException e1) {
-                                logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                         e1.getMessage());
                             }
 
@@ -1509,7 +1511,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                     }
                                 }
                             } catch (SchedulerException e) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                         e.getMessage());
                             }
@@ -1563,7 +1565,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                             theDatagramChannel.getRemoteAddress());
                                 }
                             } catch (IOException e) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while getting the remote address of channel {} ({})",
                                         theDatagramChannel, e.getMessage());
                             }
@@ -1639,11 +1641,11 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                     }
                                 } catch (IOException e) {
                                     if (theElement.channel.lastRemote != null) {
-                                        logger.error(
+                                        logger.warn(
                                                 "An exception occurred while sending data to the remote end {} ({})",
                                                 theElement.channel.lastRemote, e.getMessage());
                                     } else {
-                                        logger.error(
+                                        logger.warn(
                                                 "An exception occurred while sending data to the remote end {} ({})",
                                                 theElement.channel.remote, e.getMessage());
                                     }
@@ -1681,7 +1683,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                     try {
                                         scheduler = StdSchedulerFactory.getDefaultScheduler();
                                     } catch (SchedulerException e1) {
-                                        logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                        logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                                 e1.getMessage());
                                     }
 
@@ -1714,7 +1716,7 @@ public abstract class AbstractDatagramChannelBinding<P extends ChannelBindingPro
                                             }
                                         }
                                     } catch (SchedulerException e) {
-                                        logger.error(
+                                        logger.warn(
                                                 "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                                 e.getMessage());
                                     }

--- a/bundles/binding/org.openhab.binding.tcp/src/main/java/org/openhab/binding/tcp/AbstractSocketChannelBinding.java
+++ b/bundles/binding/org.openhab.binding.tcp/src/main/java/org/openhab/binding/tcp/AbstractSocketChannelBinding.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2010-2016 by the respective copyright holders.
+ * Copyright (c) 2010-2017 by the respective copyright holders.
  *
  * All rights reserved. This program and the accompanying materials
  * are made available under the terms of the Eclipse Public License v1.0
@@ -28,6 +28,7 @@ import java.util.Collections;
 import java.util.Dictionary;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Objects;
 
 import org.apache.commons.lang.StringUtils;
 import org.openhab.core.binding.AbstractActiveBinding;
@@ -51,14 +52,12 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- *
- * This is the base for all "Socket" connection-oriented network based connectivity and communication.It
- * requires a ChannelBindingProvider based binding provider. Data is pushed around using ByteBuffers with an indicator
- * for blocking/non-blocking (synchronous/asynchronous) communication
+ * This is the base for all "Socket" connection-oriented network based connectivity and communication. It
+ * requires a ChannelBindingProvider-based binding provider. Data is pushed around using ByteBuffers with an indicator
+ * for blocking/non-blocking (synchronous/asynchronous) communication.
  *
  * @author Karel Goderis
  * @since 1.1.0
- *
  */
 public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvider> extends AbstractActiveBinding<P>
         implements ManagedService {
@@ -77,9 +76,9 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     protected boolean queueUntilConnected = true;
     // default port to listen on for incoming connections
     protected int listenerPort = 0;
-    // share channels between within an item definition
+    // share channels within an item definition
     protected boolean itemShareChannels = true;
-    // share channels between items definitions
+    // share channels between item definitions
     protected boolean bindingShareChannels = true;
     // share channels between "outbound" and "inbound" item definitions
     protected boolean directionsShareChannels = false;
@@ -94,15 +93,15 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     // Queue to store BufferElements that need to be written to the network
     protected List<WriteBufferElement> writeQueue = Collections.synchronizedList(new ArrayList<WriteBufferElement>());
 
-    // Simple datastructure to track the state of Channels
+    // Simple data structure to track the state of Channels
     protected ChannelTracker<Channel> channels = new ChannelTracker<Channel>();
 
     /**
-     * Datastructure to represent that state of a communications channel
-     * 
+     * Data structure to represent the state of a communications channel.
+     *
      * @author Karel Goderis
      * @since 1.4.0
-     **/
+     */
     protected class Channel {
 
         // the Item that is bound to this channel
@@ -193,7 +192,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                 response = response + "]";
                 return response;
             } catch (Exception e) {
-                logger.error("An exception occurred while converting Channel to String {}", e.getMessage());
+                logger.warn("An exception occurred while converting Channel to String: {}", e.getMessage());
             }
             return "";
         }
@@ -204,7 +203,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
      * The ChannelTracker acts as a little dB that stores all the information on the state of the
      * underlying NIO SocketChannels in use. It comes with a bunch of get... methods that allow a caller to
      * query Channels.
-     * 
+     *
      * get() - get the channel that matches the provided criteria for the given {Item,Command}
      * getFirst() - get the first channel that matches the provided criteria
      * getFirstServed() - return the first Channel that matches the criteria AND that is currently bound to a Java NIO
@@ -212,11 +211,10 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
      * getAll() - return a collection of all the Channels that match the given criteria
      * contains() - return true if a channel that matches the provided criteria exists in the ChannelTracker
      * replace() - replaces the underlying Java NIO channel on the Channels that match the provided criteria
-     * 
+     *
      * @author Karel Goderis
      * @since 1.4.0
-     * 
-     **/
+     */
     protected class ChannelTracker<C extends Channel> extends ArrayList<C> {
 
         private static final long serialVersionUID = 1543958347565096785L;
@@ -631,13 +629,11 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     }
 
     /**
-     * Simple helper class to store data that needs to be sent over a given channel
-     * 
-     * 
+     * Simple helper class to store data that needs to be sent over a given channel.
+     *
      * @author Karel Goderis
      * @since 1.4.0
-     * 
-     **/
+     */
     protected class WriteBufferElement {
         public Channel channel;
         public ByteBuffer buffer;
@@ -687,12 +683,12 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                 try {
                     listenerKey = listenerChannel.register(selector, SelectionKey.OP_ACCEPT);
                 } catch (ClosedChannelException e1) {
-                    logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+                    logger.warn("An exception occurred while registering a selector: {}", e1.getMessage());
                 }
             }
 
         } catch (Exception e3) {
-            logger.error("An exception occurred while creating the Listener Channel on port number {} ({})",
+            logger.warn("An exception occurred while creating the Listener Channel on port number {}: {}",
                     listenerPort, e3.getMessage());
         }
     }
@@ -707,7 +703,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
         try {
             selector = Selector.open();
         } catch (IOException e) {
-            logger.error("An exception occurred while registering the selector: {}", e.getMessage());
+            logger.warn("An exception occurred while registering the selector: {}", e.getMessage());
         }
     }
 
@@ -720,24 +716,25 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
         try {
             selector.close();
         } catch (IOException e) {
-            logger.error("An exception occurred while closing the selector: {}", e.getMessage());
+            logger.warn("An exception occurred while closing the selector: {}", e.getMessage());
         }
 
         try {
-            listenerChannel.close();
+            if (listenerChannel != null) {
+                listenerChannel.close();
+            }
         } catch (IOException e) {
-            logger.error("An exception occurred while closing the Listener Channel on port number {} ({})",
-                    listenerPort, e.getMessage());
-
+            logger.warn("An exception occurred while closing the Listener Channel on port number {}: {}", listenerPort,
+                    e.getMessage());
         }
     }
 
     /**
      * Find the first matching {@link ChannelBindingProvider}
      * according to <code>itemName</code>
-     * 
+     *
      * @param itemName
-     * 
+     *
      * @return the matching binding provider or <code>null</code> if no binding
      *         provider could be found
      */
@@ -769,14 +766,14 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     public void updated(Dictionary config) throws ConfigurationException {
 
         if (config != null) {
-            String bufferString = (String) config.get("buffersize");
+            String bufferString = Objects.toString(config.get("buffersize"), null);
             if (StringUtils.isNotBlank(bufferString)) {
                 maximumBufferSize = Integer.parseInt((bufferString));
             } else {
                 logger.info("The maximum buffer will be set to the default value of {}", maximumBufferSize);
             }
 
-            String reconnectString = (String) config.get("retryinterval");
+            String reconnectString = Objects.toString(config.get("retryinterval"), null);
             if (StringUtils.isNotBlank(reconnectString)) {
                 reconnectInterval = Integer.parseInt((reconnectString));
             } else {
@@ -784,14 +781,14 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         reconnectInterval);
             }
 
-            String cronString = (String) config.get("reconnectcron");
+            String cronString = Objects.toString(config.get("reconnectcron"), null);
             if (StringUtils.isNotBlank(cronString)) {
                 reconnectCron = cronString;
             } else {
                 logger.info("The cron job to reset connections will be set to the default value of {}", reconnectCron);
             }
 
-            String queueString = (String) config.get("queue");
+            String queueString = Objects.toString(config.get("queue"), null);
             if (StringUtils.isNotBlank(queueString)) {
                 queueUntilConnected = Boolean.parseBoolean(queueString);
             } else {
@@ -800,7 +797,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         queueUntilConnected);
             }
 
-            String portString = (String) config.get("port");
+            String portString = Objects.toString(config.get("port"), null);
             if (StringUtils.isNotBlank(portString)) {
                 listenerPort = Integer.parseInt((portString));
             } else {
@@ -808,7 +805,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         listenerPort);
             }
 
-            String shareString = (String) config.get("itemsharedconnections");
+            String shareString = Objects.toString(config.get("itemsharedconnections"), null);
             if (StringUtils.isNotBlank(shareString)) {
                 itemShareChannels = Boolean.parseBoolean(shareString);
             } else {
@@ -816,7 +813,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         itemShareChannels);
             }
 
-            String shareString2 = (String) config.get("bindingsharedconnections");
+            String shareString2 = Objects.toString(config.get("bindingsharedconnections"), null);
             if (StringUtils.isNotBlank(shareString2)) {
                 bindingShareChannels = Boolean.parseBoolean(shareString2);
             } else {
@@ -825,7 +822,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         bindingShareChannels);
             }
 
-            String shareString3 = (String) config.get("directionssharedconnections");
+            String shareString3 = Objects.toString(config.get("directionssharedconnections"), null);
             if (StringUtils.isNotBlank(shareString3)) {
                 directionsShareChannels = Boolean.parseBoolean(shareString3);
             } else {
@@ -833,7 +830,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         bindingShareChannels);
             }
 
-            String shareString4 = (String) config.get("addressmask");
+            String shareString4 = Objects.toString(config.get("addressmask"), null);
             if (StringUtils.isNotBlank(shareString4)) {
                 useAddressMask = Boolean.parseBoolean(shareString4);
             } else {
@@ -844,24 +841,24 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
 
             if (useAddressMask && directionsShareChannels) {
                 logger.warn(
-                        "The setting to share channels between directions is not compatible with the setting to use address masks. We will override the setting to share between directions");
+                        "The setting to share channels between directions is not compatible with the setting to use address masks. We will override the setting to share between directions.");
                 directionsShareChannels = false;
             }
 
             if (bindingShareChannels && !itemShareChannels) {
                 logger.warn(
-                        "The setting to share channels in the binding is not compatible with the setting to share channels within items. We will override the setting to share between items");
+                        "The setting to share channels in the binding is not compatible with the setting to share channels within items. We will override the setting to share between items.");
                 itemShareChannels = true;
             }
 
             if (directionsShareChannels && (!bindingShareChannels || !itemShareChannels)) {
                 logger.warn(
-                        "The setting to share channels between directions is not compatible with the setting to share channels between items or within items. We will override the settings");
+                        "The setting to share channels between directions is not compatible with the setting to share channels between items or within items. We will override the settings.");
                 itemShareChannels = true;
                 bindingShareChannels = true;
             }
 
-            String refreshString = (String) config.get("refreshinterval");
+            String refreshString = Objects.toString(config.get("refreshinterval"), null);
             if (StringUtils.isNotBlank(refreshString)) {
                 refreshInterval = Long.parseLong((refreshString));
             } else {
@@ -874,9 +871,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
             }
 
             setProperlyConfigured(true);
-
         }
-
     }
 
     /**
@@ -887,7 +882,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
         P provider = findFirstMatchingBindingProvider(itemName);
 
         if (provider == null) {
-            logger.warn("cannot find matching binding provider [itemName={}, command={}]", itemName, command);
+            logger.warn("Cannot find matching binding provider [itemName={}, command={}]", itemName, command);
             return;
         }
 
@@ -922,7 +917,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                     || provider.getPortAsString(itemName, someCommand).equals("*")))) {
 
                         logger.warn(
-                                "The channel for {} has a connection problem. Data will queued to the new channel when it is successfully set up.",
+                                "The channel for {} has a connection problem. Data will be queued to the new channel when it is successfully set up.",
                                 theChannel.remote);
 
                         if (!theSocketChannel.isConnectionPending() || !theSocketChannel.isOpen()) {
@@ -931,7 +926,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                             try {
                                 scheduler = StdSchedulerFactory.getDefaultScheduler();
                             } catch (SchedulerException e1) {
-                                logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                         e1.getMessage());
                             }
 
@@ -957,7 +952,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                     }
                                 }
                             } catch (SchedulerException e) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                         e.getMessage());
                             }
@@ -974,7 +969,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         }
                     }
                 } else {
-                    logger.error("there is no channel that services [itemName={}, command={}]", itemName, command);
+                    logger.warn("There is no channel that services [itemName={}, command={}]", itemName, command);
                 }
             }
         }
@@ -982,14 +977,15 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
 
     /**
      * This function should be implemented and used to "setup" the
-     * ASCII protocol after that the socket channel has been created. This because some ASCII
-     * protocols need to first emit a certain sequence of characters to
-     * configure or setup the communication channel with the remote end
-     **/
+     * ASCII protocol after the socket channel has been created. This is
+     * because some ASCII protocols need to first emit a certain sequence
+     * of characters to configure or set up the communication channel with
+     * the remote end.
+     */
     abstract protected void configureChannel(Channel channel);
 
     /**
-     * The actual implementation for receiving a command from the openhab runtime should go here
+     * The actual implementation for receiving a command from the openHAB runtime should go here.
      *
      * @param itemName the item name
      * @param command the command
@@ -1001,13 +997,14 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
             String commandAsString);
 
     /**
-     * Returns a {@link State} which is inherited from provide list of DataTypes. The call is delegated to the
-     * {@link TypeParser}. If
-     * <code>stateTypeList</code> is <code>null</code> the {@link StringType} is used.
-     * 
+     * Returns a {@link State} which is inherited from provided list of
+     * DataTypes. The call is delegated to the {@link TypeParser}. If
+     * <code>stateTypeList</code> is <code>null</code> the {@link StringType}
+     * is used.
+     *
      * @param stateTypeList - a list of state types that we should parse against
      * @param transformedResponse - the string to be parsed
-     * 
+     *
      * @return a {@link State} which type is inherited by the {@link TypeParser}
      *         or a {@link StringType} if <code>stateTypeList</code> is <code>null</code>
      */
@@ -1020,7 +1017,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     }
 
     /**
-     * Parses the buffer received from the Channel
+     * Parses the buffer received from the Channel.
      *
      * @param networkChannel the network channel
      * @param byteBuffer the byte buffer
@@ -1032,15 +1029,14 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     }
 
     /**
-     * 
      * Callback that will be called when data is received on a given channel.
-     * This method should deal with the actual details of the protocol being implemented
+     * This method should deal with the actual details of the protocol being implemented.
      */
     abstract protected void parseBuffer(String itemName, Command aCommand, Direction theDirection,
             ByteBuffer byteBuffer);
 
     /**
-     * Queues (writes) a ByteBuffer to a channel
+     * Queues (writes) a ByteBuffer to a channel.
      *
      * @param theChannel the network channel
      * @param byteBuffer the byte buffer
@@ -1095,11 +1091,10 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     }
 
     /**
-     * Quartz Job to reconnect a channel
-     * 
+     * Quartz Job to reconnect a channel.
+     *
      * @author Karel Goderis
      * @since 1.2.0
-     *
      */
     public static class ReconnectJob implements Job {
 
@@ -1123,13 +1118,13 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                     try {
                         theChannel.channel.close();
                     } catch (IOException e) {
-                        logger.error("An exception occurred while closing a channel: {}", e.getMessage());
+                        logger.warn("An exception occurred while closing a channel: {}", e.getMessage());
                     }
 
                     try {
                         theChannel.channel = SocketChannel.open();
                     } catch (IOException e) {
-                        logger.error("An exception occurred while opening a channel: {}", e.getMessage());
+                        logger.warn("An exception occurred while opening a channel: {}", e.getMessage());
                     }
 
                     theChannel.isBlocking = false;
@@ -1139,7 +1134,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         theChannel.channel.configureBlocking(false);
                         // setKeepAlive(true);
                     } catch (Exception e) {
-                        logger.error("An exception occurred while configuring a channel: {}", e.getMessage());
+                        logger.warn("An exception occurred while configuring a channel: {}", e.getMessage());
                     }
 
                     synchronized (theBinding.selector) {
@@ -1150,7 +1145,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                 theChannel.channel.register(theBinding.selector, interestSet);
                             }
                         } catch (ClosedChannelException e1) {
-                            logger.error("An exception occurred while registering a selector: {}", e1.getMessage());
+                            logger.warn("An exception occurred while registering a selector: {}", e1.getMessage());
                         }
                     }
 
@@ -1160,23 +1155,22 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                             logger.info("Attempting to reconnect the channel for {}", theChannel.remote);
                         }
                     } catch (Exception e) {
-                        logger.error("An exception occurred while connecting a channel: {}", e.getMessage());
+                        logger.warn("An exception occurred while connecting a channel: {}", e.getMessage());
                     }
                 } else {
-                    logger.debug("I cannot proceed without remote address");
+                    logger.debug("Either the remote address was not found or the channel was not open.");
                 }
             } else {
-                logger.warn("Already reconnecting the channel for {}", theChannel.remote);
+                logger.warn("Channel for {} is not reconnecting.", theChannel.remote);
             }
         }
     }
 
     /**
-     * Quartz Job to configure a channel
-     * 
+     * Quartz Job to configure a channel.
+     *
      * @author Karel Goderis
      * @since 1.4.0
-     *
      */
     public static class ConfigureJob implements Job {
 
@@ -1283,8 +1277,8 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                             boolean assigned = false;
 
                             if (useAddressMask && (remoteHost.equals("*") || remotePort.equals("*"))) {
-                                logger.error(
-                                        "We do not accept outgoing connections for Items that do use address masks");
+                                logger.warn(
+                                        "We do not accept outgoing connections for Items that use address masks");
                             } else {
 
                                 channels.add(newChannel);
@@ -1328,7 +1322,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                             try {
                                                 newSocketChannel = SocketChannel.open();
                                             } catch (IOException e2) {
-                                                logger.error("An exception occurred while opening a channel: {}",
+                                                logger.warn("An exception occurred while opening a channel: {}",
                                                         e2.getMessage());
                                             }
 
@@ -1336,7 +1330,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                                 newSocketChannel.socket().setKeepAlive(true);
                                                 newSocketChannel.configureBlocking(false);
                                             } catch (IOException e) {
-                                                logger.error("An exception occurred while configuring a channel: {}",
+                                                logger.warn("An exception occurred while configuring a channel: {}",
                                                         e.getMessage());
                                             }
 
@@ -1347,7 +1341,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                                 try {
                                                     newSocketChannel.register(selector, interestSet);
                                                 } catch (ClosedChannelException e1) {
-                                                    logger.error(
+                                                    logger.warn(
                                                             "An exception occurred while registering a selector: {}",
                                                             e1.getMessage());
                                                 }
@@ -1360,7 +1354,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                                 logger.info("Connecting the channel {} ", newChannel);
                                                 newSocketChannel.connect(remoteAddress);
                                             } catch (IOException e) {
-                                                logger.error("An exception occurred while connecting a channel: {}",
+                                                logger.warn("An exception occurred while connecting a channel: {}",
                                                         e.getMessage());
                                             }
                                         }
@@ -1382,7 +1376,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                 // Wait for an event
                 selector.selectNow();
             } catch (IOException e) {
-                logger.error("An exception occurred while Selecting ({})", e.getMessage());
+                logger.warn("An exception occurred while Selecting: {}", e.getMessage());
             }
         }
 
@@ -1446,7 +1440,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                             newChannel.configureBlocking(false);
                                             // setKeepAlive(true);
                                         } catch (IOException e) {
-                                            logger.error("An exception occurred while configuring a channel: {}",
+                                            logger.warn("An exception occurred while configuring a channel: {}",
                                                     e.getMessage());
                                         }
 
@@ -1455,7 +1449,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                             try {
                                                 newChannel.register(selector, newChannel.validOps());
                                             } catch (ClosedChannelException e1) {
-                                                logger.error("An exception occurred while registering a selector: {}",
+                                                logger.warn("An exception occurred while registering a selector: {}",
                                                         e1.getMessage());
                                             }
                                         }
@@ -1464,7 +1458,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                         try {
                                             scheduler = StdSchedulerFactory.getDefaultScheduler();
                                         } catch (SchedulerException e1) {
-                                            logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                            logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                                     e1.getMessage());
                                         }
 
@@ -1491,7 +1485,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                                 scheduler.scheduleJob(job, trigger);
                                             }
                                         } catch (SchedulerException e) {
-                                            logger.error(
+                                            logger.warn(
                                                     "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                                     e.getMessage());
                                         }
@@ -1514,7 +1508,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                             }
 
                         } catch (IOException e) {
-                            logger.error("An exception occurred while configuring a channel: {}", e.getMessage());
+                            logger.warn("An exception occurred while configuring a channel: {}", e.getMessage());
                         }
                     }
                 } else {
@@ -1552,7 +1546,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                             try {
                                 scheduler = StdSchedulerFactory.getDefaultScheduler();
                             } catch (SchedulerException e1) {
-                                logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                         e1.getMessage());
                             }
 
@@ -1578,7 +1572,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                     }
                                 }
                             } catch (SchedulerException e) {
-                                logger.error(
+                                logger.warn(
                                         "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                         e.getMessage());
                             }
@@ -1589,7 +1583,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                 try {
                                     remote = (InetSocketAddress) theSocketChannel.getRemoteAddress();
                                 } catch (IOException e) {
-                                    logger.error(
+                                    logger.warn(
                                             "An exception occurred while getting the remote address of channel {} ({})",
                                             theSocketChannel, e.getMessage());
                                 }
@@ -1612,7 +1606,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                 try {
                                     scheduler = StdSchedulerFactory.getDefaultScheduler();
                                 } catch (SchedulerException e1) {
-                                    logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                    logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                             e1.getMessage());
                                 }
 
@@ -1635,7 +1629,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                         scheduler.scheduleJob(job, trigger);
                                     }
                                 } catch (SchedulerException e) {
-                                    logger.error(
+                                    logger.warn(
                                             "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                             e.getMessage());
                                 }
@@ -1655,7 +1649,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                         scheduler.scheduleJob(job, trigger);
                                     }
                                 } catch (SchedulerException e) {
-                                    logger.error(
+                                    logger.warn(
                                             "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                             e.getMessage());
                                 }
@@ -1669,8 +1663,8 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                         boolean error = false;
 
                         try {
-                            // TODO: Additional code to split readBuffer in multiple parts, in case the data send by the
-                            // remote end is not correctly fragemented. Could be handed of to implementation class if
+                            // TODO: Additional code to split readBuffer into multiple parts, in case the data sent by the
+                            // remote end is not correctly fragmented. Could be handed off to implementation class if,
                             // for example, the buffer needs to be split based on a special character like line feed or
                             // carriage return
                             numberBytesRead = theSocketChannel.read(readBuffer);
@@ -1691,7 +1685,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                             try {
                                 theSocketChannel.close();
                             } catch (IOException e) {
-                                logger.warn("The channel for {} is closed ({})", theChannel.remote, e.getMessage());
+                                logger.warn("Error occurred while closing the channel for {}: {}", theChannel.remote, e.getMessage());
                             }
                             error = true;
                         }
@@ -1703,7 +1697,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                 try {
                                     scheduler = StdSchedulerFactory.getDefaultScheduler();
                                 } catch (SchedulerException e1) {
-                                    logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                    logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                             e1.getMessage());
                                 }
 
@@ -1731,7 +1725,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                         }
                                     }
                                 } catch (SchedulerException e) {
-                                    logger.error(
+                                    logger.warn(
                                             "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                             e.getMessage());
                                 }
@@ -1772,7 +1766,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                             "No channel is active or defined for the data we received from {}. It will be discarded.",
                                             theSocketChannel.getRemoteAddress());
                                 } catch (IOException e) {
-                                    logger.error(
+                                    logger.warn(
                                             "An exception occurred while getting the remote address of the channel {} ({})",
                                             theSocketChannel, e.getMessage());
                                 }
@@ -1843,7 +1837,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                         try {
                                             scheduler = StdSchedulerFactory.getDefaultScheduler();
                                         } catch (SchedulerException e1) {
-                                            logger.error("An exception occurred while getting the Quartz scheduler: {}",
+                                            logger.warn("An exception occurred while getting the Quartz scheduler: {}",
                                                     e1.getMessage());
                                         }
 
@@ -1871,7 +1865,7 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
                                                 }
                                             }
                                         } catch (SchedulerException e) {
-                                            logger.error(
+                                            logger.warn(
                                                     "An exception occurred while scheduling a job with the Quartz Scheduler {}",
                                                     e.getMessage());
                                         }
@@ -1900,5 +1894,4 @@ public abstract class AbstractSocketChannelBinding<P extends ChannelBindingProvi
     protected long getRefreshInterval() {
         return refreshInterval;
     }
-
 }

